docs/source/features/activation_recomputation.rst

```diff
@@ -0,0 +1,33 @@
+Activation Recomputation
+========================
+
+The input activations of network layers are stored in the device memory to compute the gradients in back-propagation.
+The input activation stores easily saturate the device memory when training a LLM with a large sequence length or a large micro-batch size.
+Check-pointing a few activations and recomputing the rest of activations is a common technique to reduce the need of device memory.
+
+Transformer Layer Recomputation
+-------------------------------
+
+NeMo supports Transformer layer recomputation that checkpoints the input of each Transformer layer and recomputes the activations of the rest of layers.
+Transformer layer recomputation significantly reduces the activation memory usage.
+However, this approach increases per-Transformer layer computation cost by 30%, which comes from re-executing the entire layer forwarding computation.
+NeMo also supports partial Transformer layer recomputation, which is beneficial when recomputing a few Transformer layers would fit the training workload on GPU memory.
+This would avoid recomputing the rest of layers.
```

Comment:
NeMo supports Transformer layer recomputation, which involves checkpointing the input of each Transformer layer and recomputing the activations for the remaining layers. This technique significantly reduces activation memory usage. However, it does come with a 30% increase in per-Transformer layer computation cost due to re-executing the entire layer forwarding computation. Additionally, NeMo also provides support for partial Transformer layer recomputation. This can be beneficial when recomputing only a few Transformer layers fits within the available GPU memory, avoiding the need to recompute the remaining layers.